src/config.rs

```diff
@@ -181,9 +169,115 @@ impl Configuration {
         Ok(torrust_config)
     }
 
-    pub fn save_to_file(&self) -> Result<(), ()> {
+    pub fn save_to_file(&self, path: &str) -> Result<(), ()> {
         let toml_string = toml::to_string(self).expect("Could not encode TOML value");
-        fs::write("config.toml", toml_string).expect("Could not write to file!");
+        fs::write(path, toml_string).expect("Could not write to file!");
         Ok(())
     }
 }
+
+mod configuration {
```

Comment:
I think the module for tests should be just named `mod tests` regardless of what the name of the to be tested module is. The entire test module should then be annotated with `#[cfg(test)]`. All test functions in this module should then be annotated with `#[test]`.

Eg:
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn test_func1() {
        assert!(true);
    }

    #[test]
    fn test_func2() {
        assert!(true);
    }
}
```

https://doc.rust-lang.org/book/ch11-03-test-organization.html